### PR TITLE
CVE-2014-0482 and CVE-2018-14574

### PR DESCRIPTION
**Pull request by sjc5897**
Small Django vulnerability in their middleware that allowed remote users to receive other user's session.

### CODE REVIEW

**Comment by sjc5897** [comment]
I have 2 django vulnerabilities, I thought it was one django one tomcat

File: cves/CVE-2014-0482.yml [hunk 1]
```diff
@@ -15,24 +15,24 @@ curated_instructions: |
...
   set to true.
-curated: false
+curated: true
 reported_instructions: |
   What date was the vulnerability reported to the security team? Look at the
   security bulletins and bug reports. It is not necessarily the same day that the
   CVE was created.  Leave blank if no date is given.
   Please enter your date in YYYY-MM-DD format.
-reported_date:
+reported_date: 2014-07-21
```
Comment (jmc9961) [inline cves/CVE-2014-0482.yml:24]: Would you mind clarifying where you found this date?

File: cves/CVE-2014-0482.yml [hunk 2]
```diff
@@ -270,7 +295,19 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This appears to me to be a coding mistake. It would appear that this
+    vulnerability was created as misunderstanding of the interaction between the
```
Comment (jmc9961) [inline cves/CVE-2014-0482.yml]: There should be an 'a' between 'as' and 'misunderstanding'

File: cves/CVE-2014-0482.yml [hunk 3]
```diff
@@ -270,7 +295,19 @@ mistakes:
...
+    middleware and backend authentication. It could also be argued that this is a
+    design mistake; however, I have been unable to find designs to show this.
```
Comment (jmc9961) [inline cves/CVE-2014-0482.yml]: Could you clarify on how you would argue that it is also a design mistake here?

File: cves/CVE-2018-14574.yml [hunk 4]
```diff
@@ -170,8 +177,11 @@ autodiscoverable:
 
     The answer field should be boolean. In answer_note, please explain
     why you come to that conclusion.
-  answer_note:
-  answer:
+  answer_note: |
+    I think that this could be discovered by fuzzer looking for Open Redirect and
+    sending internationally malformed urls; however, you may need a very specific
```
Comment (jmc9961) [inline cves/CVE-2018-14574.yml]: Did you mean to say 'intentionally' instead of 'internationally'?

File: cves/CVE-2018-14574.yml [hunk 5]
```diff
@@ -266,8 +285,11 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: |
+      Part of the issue that caused this security vulnrability elimnated some input validation.
```
Comment (jmc9961) [inline cves/CVE-2018-14574.yml]: 'vulnerability' and 'eliminated' are misspelled.

File: cves/CVE-2018-14574.yml [hunk 6]
```diff
@@ -298,7 +320,17 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    I would say that this is a mistake in the Requirements of the project. This
+    bug was introduced via an attempted optimization. This optimization seemingly
+    eliminated an important portion of the url validation. In addition, this shows
+    some of the issues of an open source project. The person who reported the initial
+    optimization opportunity was also the person who commit the optimization and thus
```
Comment (jmc9961) [inline cves/CVE-2018-14574.yml]: 'commit' should be 'committed'

**Review by jmc9961 — COMMENTED** [review]
I'm not sure exactly how the upvotes are supposed to work, so I guess I will give one to each?

File: cves/CVE-2014-0482.yml [hunk 7]
```diff
@@ -142,8 +152,9 @@ autodiscoverable:
 
     The answer field should be boolean. In answer_note, please explain
     why you come to that conclusion.
-  answer_note:
-  answer:
+  answer_note: |
+    This vulnerability requires understanding of authentication requirements for RemoteUserMiddleware
```
Comment (dad9489) [inline cves/CVE-2014-0482.yml]: It is not clear why this understanding would prevent an automatic tool from detecting this as long as it was designed by someone with this understanding. Maybe elaborate?

File: cves/CVE-2018-14574.yml [hunk 8]
```diff
@@ -49,7 +49,11 @@ description_instructions: |
 
   Your target audience is people just like you before you took any course in
   security
-description:
+description: |
+  There is a vulnerability that causes Open redirects in the django middleware.
+  This vulnerability occurs when certain settings are enabled and the URL pattern
+  accepts paths ending in slash. This allows attackers to create maliciously crafted
+  URL that can lead to redirects to other sites.
```
Comment (dad9489) [inline cves/CVE-2018-14574.yml]: This should either say " a maliciously crafted URL" or "maliciously crafted URLs"

File: cves/CVE-2018-14574.yml [hunk 9]
```diff
@@ -135,10 +139,10 @@ unit_tested:
 
     For the fix_answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  code:
-  code_answer:
-  fix:
-  fix_answer:
+  code: true
+  code_answer: This file had unit tests; however, no unit test were created to catch this edgecase
```
Comment (dad9489) [inline cves/CVE-2018-14574.yml]: should be " no unit tests" and "edge case"

File: cves/CVE-2018-14574.yml [hunk 10]
```diff
@@ -298,7 +320,17 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    I would say that this is a mistake in the Requirements of the project. This
```
Comment (dad9489) [inline cves/CVE-2018-14574.yml:324]: I don't understand what you mean by this being a mistake in the requirements. I get that this was introduced by an optimization that was not properly reviewed, but I don't see how that is caused by the requirements. Maybe explain a bit more what you mean by this?

**Review by dad9489 — CHANGES_REQUESTED** [review]

File: cves/CVE-2014-0482.yml [hunk 11]
```diff
@@ -283,8 +320,7 @@ CWE_instructions: |
 
   Just the number here is fine. No need for name or CWE prefix. If more than one
   apply here, then choose the best one and mention the others in CWE_note.
-CWE:
-CWE_note:
+CWE: 1211
```
Comment (dad9489) [inline cves/CVE-2014-0482.yml:323]: Upvotes: 2

File: cves/CVE-2018-14574.yml [hunk 12]
```diff
@@ -311,8 +343,8 @@ CWE_instructions: |
 
   Just the number here is fine. No need for name or CWE prefix. If more than one
   apply here, then choose the best one and mention the others in CWE_note.
-CWE:
-CWE_note:
+CWE: 601
```
Comment (dad9489) [inline cves/CVE-2018-14574.yml:347]: Upvotes: 3

File: cves/CVE-2014-0482.yml (hunk 1 above)
Comment (sjc5897) [reply cves/CVE-2014-0482.yml:24]: https://code.djangoproject.com/ticket/23066 I was able to find a django ticket. The ticket number was found in a commit message.
